sound/soc/sof/intel/pci-mtl.c

```diff
@@ -50,9 +50,40 @@ static const struct sof_dev_desc mtl_desc = {
 	.ops_free = hda_ops_free,
 };
 
+static const struct sof_dev_desc arl_desc = {
+        .use_acpi_target_states = true,
+        .machines               = snd_soc_acpi_intel_arl_machines,
+        .alt_machines           = snd_soc_acpi_intel_arl_sdw_machines,
+        .resindex_lpe_base      = 0,
+        .resindex_pcicfg_base   = -1,
+        .resindex_imr_base      = -1,
+        .irqindex_host_ipc      = -1,
+       .chip_info = &mtl_chip_info,
+        .ipc_supported_mask     = BIT(SOF_INTEL_IPC4),
+        .ipc_default            = SOF_INTEL_IPC4,
```

Comment:
unfortunately this conflicts with the renaming added by @ujfalusi with "ASoC: SOF: Intel: Use generic names for IPC types"

So the net result is that this PR will not build and we can't send this upstream - and it will be difficult to apply to stable kernels.

yay renaming.